### PR TITLE
feat: add oneclick for ccm

### PR DESCRIPTION
**Pull request by GaryJBlake**
### Summary

- Added `Export-CcmJsonSpec` cmdlet to generate a JSON specification file for Cross Cloud Migration.
- Added `Invoke-CcmDeployment` cmdlet to perform an end-to-end deployment of Cross Cloud Migration.
- Added `Invoke-UndoCcmDeployment` cmdlet to perform removal of Cross Cloud Migration.
- Added documentation for new functions.

### Type

- [ ] Bugfix
- [x] Enhancement or Feature
- [ ] Code Style or Formatting
- [x] Documentation
- [ ] Refactoring
- [ ] Chore
- [ ] Other
        Please describe:

### Breaking Changes?

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### Test and Documentation

- [x] Tests have been completed.
- [x] Documentation has been added or updated.

#### Exporting the JSON

image

#### Deployment

image

#### Removal

image

### Issue References

N/A

### Additional Information

N/A

### CODE REVIEW

File: docs/documentation/functions/solutions/ccm/Invoke-CcmDeployment.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Invoke-CcmDeployment
+
+## SYNOPSIS
```
Comment (tenthirtyam) [inline docs/documentation/functions/solutions/ccm/Invoke-CcmDeployment.md:3]:
```suggestion
## Synopsis

```

File: docs/documentation/functions/solutions/ccm/Invoke-UndoCcmDeployment.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+## Examples
+
+### Example 1
+
+``` powershell
+Invoke-UndoCcmDeployment -jsonFile .\ccmDeploySpec.json
+```
+
+This example removes CCross Cloud Mobility for VMware Cloud Foundation using the JSON spec supplied.
```
Comment (tenthirtyam) [inline docs/documentation/functions/solutions/ccm/Invoke-UndoCcmDeployment.md:26]:
```suggestion
This example removes Cross Cloud Mobility for VMware Cloud Foundation using the JSON spec supplied.

```

File: PowerValidatedSolutions.psm1 [hunk 3]
```diff
@@ -22197,6 +22197,278 @@ Export-ModuleMember -Function Invoke-UndoCbrDeployment
...
+        .SYNOPSIS
+        End-to-end removal of Cross Cloud Mobility
+
+        .DESCRIPTION
+        The Invoke-UndoCcmDeployment cmdlet is a single function to removal the configuration of the Cross Cloud
+        Mobility for VMware Cloud Foundation validated solution.
+
+        .EXAMPLE
+        Invoke-UndoCcmDeployment -jsonFile .\ccmDeploySpec.json
+        This example removes CCross Cloud Mobility for VMware Cloud Foundation using the JSON spec supplied.
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:22388]:
```suggestion
        This example removes Cross Cloud Mobility for VMware Cloud Foundation using the JSON spec supplied.
```

File: PowerValidatedSolutions.psm1 [hunk 4]
```diff
@@ -22197,6 +22197,278 @@ Export-ModuleMember -Function Invoke-UndoCbrDeployment
...
+    Try {
+        Show-PowerValidatedSolutionsOutput -type NOTE -message "Starting Deployment of $solutionName"
+        if (Test-Path -Path $jsonFile) {
+            $jsonInput = (Get-Content -Path $jsonFile) | ConvertFrom-Json
+                if (Test-VCFConnection -server $jsonInput.sddcManagerFqdn ) {
+                    if (Test-VCFAuthentication -server $jsonInput.sddcManagerFqdn -user $jsonInput.sddcManagerUser -pass $jsonInput.sddcManagerPass) {
+                        $failureDetected        = $false
+                        $allWorkloadDomains     = Get-VCFWorkloadDomain
+                        $pvsModulePath          = (Get-InstalledModule -Name PowerValidatedSolutions).InstalledLocation
+                        if ((Get-VCFManager -version) -ge 5.1.0.0) {
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:22311]: Not 5.0.0.0?

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
Minor fixes and one question.

File: PowerValidatedSolutions.psm1 (hunk 4 above)
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:22311]: Don't beleive we had an issue wih 5.0.0.0 and vSphere 8 u1

**Review by tenthirtyam — COMMENTED** [review]
LGTM

**Pull request merged**